packages/runtime-core/src/apiCreateApp.ts

```diff
@@ -320,9 +322,22 @@ export function createAppAPI<HostElement>(
         }
       },
 
+      onUnmount(cleanupFn: () => void) {
+        if (typeof cleanupFn === 'function') pluginCleanupFns.push(cleanupFn)
```

Comment:
I think it's ok to just push it, and move the function type check to the warning check below, i,e, `__DEV__ && typeof cleanupFn !== 'function'`.

Reply:
Done.